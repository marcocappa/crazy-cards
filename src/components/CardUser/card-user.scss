.cray-card-user {
    border-radius: 1rem;
    border: 0.0625rem solid #e2e5e8;
    box-sizing: border-box;
    padding: 1rem;
    display: inline-block;
    background-color: white;

    &__name {
        margin: 0;
        font-size: 1.5rem;
    }

    &__age {
        font-size: 0.875rem;
        font-weight: 300;
    }

    &__details {
        display: flex;

        & div:last-child {
            border-left: 0.0625rem solid #e2e5e8;
            margin-left: 1.5rem;
            padding-left: 1.5rem;
        }
    }
}
