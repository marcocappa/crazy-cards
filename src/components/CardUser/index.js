import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';
const CardUser = ({
  title,
  firstname,
  lastname,
  dateOfBirth,
  employmentStatus,
  annualIncome,
  houseNumber,
  postcode,
}) => (
  <div className="cray-card-user">
    <p className="cray-card-user__name">
      {title} {firstname} {lastname}
    </p>
    <p> {format(dateOfBirth, 'dd-MM-yyyy')} </p>
    <div className="cray-card-user__details">
      <div>
        <p>{employmentStatus}</p>
        <p>Annual income: {annualIncome}</p>
      </div>
      <div>
        <p>House number: {houseNumber}</p>
        <p>Postcode: {postcode}</p>
      </div>
    </div>
  </div>
);

CardUser.propTypes = {
  title: PropTypes.string.isRequired,
  firstname: PropTypes.string.isRequired,
  lastname: PropTypes.string.isRequired,
  dateOfBirth: PropTypes.instanceOf(Date).isRequired,
  employmentStatus: PropTypes.string.isRequired,
  annualIncome: PropTypes.number.isRequired,
  houseNumber: PropTypes.number.isRequired,
  postcode: PropTypes.string.isRequired,
};

export default CardUser;
