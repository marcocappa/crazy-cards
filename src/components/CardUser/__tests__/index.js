import React from 'react';
import { shallow } from 'enzyme';
import CardUser from '../index';

describe('<CardUser />', () => {
  const props = {
    title: 'Mr.',
    firstname: 'Tom',
    lastname: 'Brady',
    dateOfBirth: new Date(2021, 2, 8),
    employmentStatus: 'Student',
    annualIncome: 5000,
    houseNumber: 310,
    postcode: 'E2 8NP',
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<CardUser {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
