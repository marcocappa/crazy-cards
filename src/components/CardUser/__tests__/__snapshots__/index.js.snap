// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardUser /> should match snapshot 1`] = `
<div
  className="cray-card-user"
>
  <p
    className="cray-card-user__name"
  >
    Mr.
     
    Tom
     
    Brady
  </p>
  <p
    className="cray-card-user__age"
  >
    08-03-2021
  </p>
  <div
    className="cray-card-user__details"
  >
    <div>
      <p>
        Student
      </p>
      <p>
        Annual income: 
        5000
      </p>
    </div>
    <div>
      <p>
        House number: 
        310
      </p>
      <p>
        Postcode: 
        E2 8NP
      </p>
    </div>
  </div>
</div>
`;
